### ISSUE TITLE
Star Wars: KotOR [Star Wars: Knights of the Old Republic] [4C410003]

### ISSUE THREAD

**Issue by Voxel9** [Cxbx-Reloaded/game-compatibility#428]
In the master branch, the game only gets as far as booting to a black screen with a constant whooshing sound effect (I can't really describe it any better than that), but this can only be achieved with hardware YUV overlays off, otherwise it just displays a "could not create overlay surface" error.

In Patrick's LessVertexPatching branch, the game just outright crashes with the following error, with or without YUV overlays enabled:
```
---------------------------
Cxbx-Reloaded
---------------------------
Received Exception Code 0xC0000005 @ EIP := 0x003A9534(=IDirectSoundStream_SetPosition_UNPATCHED+0x54e)

  Press "OK" to terminate emulation.
  Press "Cancel" to debug.
---------------------------
OK   Cancel   
---------------------------
```

---
[Xbe.txt](https://github.com/Cxbx-Reloaded/game-compatibility/files/1447593/Xbe.txt)
[KrnlDebug.txt](https://github.com/Cxbx-Reloaded/game-compatibility/files/1447594/KrnlDebug.txt)

---
**Previous Behaviour:**
None reported.

**Current behavior:**
Mixed...... without YUV overlay; it just boots to black screen with sound issues, and with YUV, it just crashes.

**Comment by hotzenplotz3000** [comment]
Tested with current build d57aa2be.
It plays the intro and hangs on a black screen.
The intro runs too fast and has choppy sound.
When you press a button it crashes:
```
---------------------------
Cxbx-Reloaded
---------------------------
Received Exception Code 0xC0000005 @ EIP := 0x040734E7

  Press "OK" to terminate emulation.
  Press "Cancel" to debug.
---------------------------
OK   Abbrechen   
---------------------------
```
[KrnlDebug.txt](https://github.com/Cxbx-Reloaded/game-compatibility/files/1756789/KrnlDebug.txt)

![2018-02-25 22_56_57-cxbx-reloaded d57aa2be feb 24 2018 _ emulating star wars_ kotor - fps_ 2 00 m](https://user-images.githubusercontent.com/30478365/36647251-de1dc6ca-1a82-11e8-83d5-81c804567766.png)

**Comment by hotzenplotz3000** [comment]
Reaches main menu with build 8b206cd6, but with broken graphics.
Launching new game will crash the game:
```
---------------------------
Cxbx-Reloaded
---------------------------
Received Exception Code 0xC0000005 @ EIP := 0x6FC426C5

  Press "OK" to terminate emulation.
  Press "Cancel" to debug.
---------------------------
OK   Abbrechen   
---------------------------
```
[KrnlDebug_8b206cd6.txt](https://github.com/Cxbx-Reloaded/game-compatibility/files/1764698/KrnlDebug_8b206cd6.txt)


![2018-02-27 22_03_16-cxbx-reloaded 8b206cd6 feb 27 2018 _ emulating star wars_ kotor - fps_ 41 75](https://user-images.githubusercontent.com/30478365/36754807-68eb62ac-1c0a-11e8-9f30-e8f48f84cf51.png)

**Comment by ratkosrb** [comment]
I tried this game on the development release. The intro video plays perfectly, but when it gets to the main menu it looks like this.
![untitled](https://user-images.githubusercontent.com/35845488/44637852-503e3500-a9bc-11e8-8186-9e1dcd2a4145.jpg)

**Comment by pederdm000** [comment]
every time i try to start a new game 
![swke](https://user-images.githubusercontent.com/20587711/46322283-cfff8680-c59c-11e8-8c0c-dc35051cc7bf.PNG)

**Comment by gandalfthewhite19890404** [comment]
Menu works fine, but sime times it just invisible. Ingame - often flickers but sometimes you can get graphic without any issues.
![image](https://user-images.githubusercontent.com/24764335/60770112-ab66ab00-a0e8-11e9-9c9d-b8606671dc5e.png)
![image](https://user-images.githubusercontent.com/24764335/60770116-b1f52280-a0e8-11e9-8dff-3907f6d60fe2.png)
![image](https://user-images.githubusercontent.com/24764335/60770140-ea94fc00-a0e8-11e9-86ba-4fc7c018a8b1.png)

**Comment by ratkosrb** [comment]
If I run it without "render directly to host backbuffer" everything except the interface is black.
![blackness](https://user-images.githubusercontent.com/35845488/61246504-a5f80900-a757-11e9-8b6d-df2a60adb55b.jpg)

**Comment by gandalfthewhite19890404** [comment]
You should use latest develop build, you don't need any hacks anymore.

**Comment by ratkosrb** [comment]
Tested again on version 688238c4 (Aug 3 2019) and the screen isn't constantly black when you get in game, but it is flickering. Like one frame is fine, then its black, then its fine again.

**Comment by ratkosrb** [comment]
Tested again and it has improved a lot. The game no longer crashes when loading saves, and there are no more invisible npcs. However these issues still remain:

- Sound cuts out at the end of the sentence when characters are speaking.
- When you finish the first level instead of continuing on Taris, you spawn in an empty room with no exit, with an entity called CutStart. The game cannot be continued beyond this point.

![cxbx 2019-12-18 03-31-11-49](https://user-images.githubusercontent.com/35845488/71048737-e44f8500-2148-11ea-947f-5804a0f1da9a.jpg)

**Comment by ratkosrb** [comment]
New feedback. Tested on latest version and loading the second level worked this time, but I had a new issue happen. As I am playing, sometimes character models randomly "explode" and look like this until the next loading screen.

![cxbx 2020-01-19 16-32-38-42](https://user-images.githubusercontent.com/35845488/72682957-97f6bc80-3adb-11ea-89d5-2c071fa6616a.jpg)

![cxbx 2020-01-19 16-40-26-48](https://user-images.githubusercontent.com/35845488/72682958-9c22da00-3adb-11ea-8213-2c6388894dd2.jpg)

**Comment by KBouder** [comment]
Game looks and runs great. But will sometimes just softlock and never load voice audio for some reason. Haven't been able to get past this point with multiple attempts.
![image](https://user-images.githubusercontent.com/11525691/81454424-3b86de00-915a-11ea-9a4c-e0abe7c79ca7.png)

**Comment by ObiKKa** [comment]
Wow. The screens in several old and recent posts above have a strange version number in the window bar. It probably doesn't need to be inserted into the issue's heading, but it's v1.95126529.

Title ID is [LA-003].

**Comment by ratkosrb** [comment]
I tried this game again and the menus are finally fixed! You can read all the text now.
Commit: dc5f56222ed037c50cbc81c2f3c5c663f011d2ee

![cxbxr-ldr 2020-11-17 02-18-04-35](https://user-images.githubusercontent.com/35845488/99325429-054fb700-287f-11eb-8097-967b49a382cf.jpg)

![cxbxr-ldr 2020-11-17 02-19-17-65](https://user-images.githubusercontent.com/35845488/99325448-0ed91f00-287f-11eb-827b-5b08331258fc.jpg)

Overall it runs very smoothly, I didn't notice any gamebreaking issues. There is one regression though, Carth is no longer visible in the scene that plays right after you enter the first corridor, where he sends a message from the bridge.

![cxbxr-ldr 2020-11-17 02-22-40-65](https://user-images.githubusercontent.com/35845488/99325599-61b2d680-287f-11eb-9c91-221a8f9deec2.jpg)

Same issue a few minutes later when he contacts you again.

![cxbxr-ldr 2020-11-17 02-30-23-12](https://user-images.githubusercontent.com/35845488/99325683-83ac5900-287f-11eb-9a1b-2ff832407c5c.jpg)

And another thing I noticed is that it asks you to reconnect the controller after every loading screen, that wasn't happening before.

![cxbxr-ldr 2020-11-17 02-40-35-09](https://user-images.githubusercontent.com/35845488/99325731-99218300-287f-11eb-8bef-a6bfa543645f.jpg)

But these small regressions aside, the game has improved tremendously, big congratulations to the CXBX-Reloaded team!

**Comment by ObiKKa** [comment]
Yes, the last lengthy comment above is due to [PR # 1982](https://github.com/Cxbx-Reloaded/Cxbx-Reloaded/pull/1982) and # 1981, dated October 11, 2020 & October 11 respectively, which fixed some problems. There is a blog post about it too.
Thanks for showing the screens and report here.
